pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java

```diff
@@ -331,7 +331,7 @@ protected int[] transformToIntValuesSVUsingValue(ValueBlock valueBlock) {
       }
     }
     if (!unselectedDocs.isEmpty()) {
-      if (_elseStatement == null) {
+      if (_elseStatement == null || LiteralTransformFunction.isNullLiteralTransform(_elseStatement)) {
```

Comment:
To simplify this, we can change `constructStatementListCalcite()` (also change `constructStatementListLegacy()` for backward compatible) to set `_elseStatement` to `null` when it is null literal